bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -255,6 +255,70 @@ def _parseline(self, line):
         raise NotImplementedError
 
 
+class InstrumentTXTResultsFileParser(InstrumentResultsFileParser):
+
+    def __init__(self, infile, separator, encoding=None,):
+        InstrumentResultsFileParser.__init__(self, infile, 'TXT')
+        # Some Instruments can generate files with different encodings, so we
+        # may need this parameter
+        self._separator = separator
+        self._encoding = encoding
+
+    def parse(self):
+        infile = self.getInputFile()
+        self.log("Parsing file ${file_name}", mapping={"file_name":infile.filename})
+        jump = 0
+        # We test in import functions if the file was uploaded
+        try:
+            if self._encoding:
+                import codecs
+                f = codecs.open(infile.name, 'r', encoding=self._encoding)
+            else:
+                f = open(infile.name, 'rU')
+        except AttributeError:
+            f = infile
+        for line in f.readlines():
```

Comment:
This file never gets closed. I'm not sure how this affects Memory in the long run, so I would prefer to use a context manager here, e.g. `with file.open(...)` and read the lines somewhere into a variable. Probably best to do that in a separate method. e.g. `lines = self.read(infile.name)`. There you could also preprocess the lines, e.g. with `lines.strip` or similar